Dockerfile.windows

```diff
@@ -94,11 +105,11 @@ RUN cmake --build . --config Release; `
 # Build OT collector
 ###############################################################################
 
-WORKDIR /work/submodules/opentelemetry-operations-collector
+FROM base as gobuilder
```

Comment:
I was initially confused why you named this `gobuilder` and not, e.g., `otcollector`, but then realized that you're reusing the same container to build all of the Go stuff. I wonder, though — why not have two builder containers, one for the OT collector, and one for the ops agent code? That way we can rebuild the ops agent without rebuilding the OT collector if needed.

Reply:
Building the OT collector is fairly fast (O(1m)) so I structured it this way so they could benefit from sharing a common Go build cache. We could certainly split the collector out.

Reply:
If it's cached at the container level, that's another minute saved for the build, and the OT collector changes very infrequently, so it's very likely to be cached… In any case, it would be nice to have a comment along the lines of "build all Go stuff in the same container", else others might be confused as I was.